integration_tests/e2e/bookingExtension.cy.ts

```diff
@@ -0,0 +1,75 @@
+import premisesFactory from '../../server/testutils/factories/premises'
+import bookingFactory from '../../server/testutils/factories/booking'
+
+import Page from '../pages/page'
+import BookingExtensionsConfirmation from '../pages/booking/extension/confirmation'
+import BookingExtensionCreatePage from '../pages/booking/extension/create'
+
+context('BookingExtension', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubAuthUser')
+  })
+
+  it('should show booking extension form', () => {
+    const booking = bookingFactory.build({
+      expectedDepartureDate: new Date(Date.UTC(2022, 5, 3, 0, 0, 0)).toISOString(),
+    })
+    const newDepartureDate = new Date(Date.UTC(2022, 6, 3, 0, 0, 0)).toISOString()
+    const premises = premisesFactory.build()
+
+    cy.task('stubBookingExtensionCreate', { premisesId: premises.id, booking })
+    cy.task('stubBookingGet', { premisesId: premises.id, booking })
+    cy.task('stubSinglePremises', { premisesId: premises.id, booking })
+
+    // Given I am signed in
+    cy.signIn()
+
+    // When I visit the booking extension page
+    const page = BookingExtensionCreatePage.visit(premises.id, booking.id)
+
+    // And I fill in the extension form
+    page.completeForm(newDepartureDate)
+    page.clickSubmit()
+
+    // Then I should be redirected to the confirmation page and the extension should be created in the API
```

Comment:
I prefer each step to be expressed as a different comment before a code block, so this should be:

```
// Then I should be redirected to the confirmation page
```

And line 40 should be:

```
// And the extension should be created in the API
```